backend/src/schema/resolvers/searches.spec.js

```diff
@@ -0,0 +1,444 @@
+import Factory, { cleanDatabase } from '../../db/factories'
+import { gql } from '../../helpers/jest'
+import { getNeode, getDriver } from '../../db/neo4j'
+import createServer from '../../server'
+import { createTestClient } from 'apollo-server-testing'
+
+let query, authenticatedUser, user
+
+const driver = getDriver()
+const neode = getNeode()
+
+beforeAll(async () => {
+  await cleanDatabase()
+  const { server } = createServer({
+    context: () => {
+      return {
+        driver,
+        neode,
+        user: authenticatedUser,
+      }
+    },
+  })
+  query = createTestClient(server).query
+})
+
+afterAll(async () => {
+  await cleanDatabase()
+})
+
+const searchQuery = gql`
+  query($query: String!) {
+    findResources(query: $query, limit: 5) {
+      __typename
+      ... on Post {
+        id
+        title
+        content
+      }
+      ... on User {
+        id
+        slug
+        name
+      }
+    }
+  }
+`
+describe('resolvers/searches', () => {
+  let variables
+
+  describe('given one user', () => {
+    beforeAll(async () => {
+      user = await Factory.build('user', {
+        id: 'a-user',
+        name: 'John Doe',
+        slug: 'john-doe',
+      })
+      authenticatedUser = await user.toJson()
+    })
+
+    describe('query contains first name of user', () => {
+      it('finds the user', async () => {
+        variables = { query: 'John' }
+        await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+          data: {
+            findResources: [
+              {
+                id: 'a-user',
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-17T09:35:27Z" title="Tuesday, March 17th 2020, 10:35:27 am +01:00">Mar 17, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,452 @@
+import Factory, { cleanDatabase } from '../../db/factories'
+import { gql } from '../../helpers/jest'
+import { getNeode, getDriver } from '../../db/neo4j'
+import createServer from '../../server'
+import { createTestClient } from 'apollo-server-testing'
+
+let query, authenticatedUser
+
+const driver = getDriver()
+const neode = getNeode()
+
+beforeAll(async () => {
+  await cleanDatabase()
+  const { server } = createServer({
+    context: () => {
+      return {
+        driver,
+        neode,
+        user: authenticatedUser,
+      }
+    },
+  })
+  query = createTestClient(server).query
+})
+
+afterAll(async () => {
+  await cleanDatabase()
+})
+
+const searchQuery = gql`
+  query($query: String!) {
+    findResources(query: $query, limit: 5) {
+      __typename
+      ... on Post {
+        id
+        title
+        content
+      }
+      ... on User {
+        id
+        slug
+        name
+      }
+    }
+  }
+`
+let user
+
+describe('resolvers', () => {
+  describe('searches', () => {
+    let variables
+
+    describe('given one user', () => {
+      beforeAll(async () => {
+        user = await Factory.build('user', {
+          id: 'a-user',
+          name: 'John Doe',
+          slug: 'john-doe',
+        })
+        authenticatedUser = await user.toJson()
+      })
+
+      const factoryOptions = {
+        authorId: 'a-user',
+      }
+
+      describe('query contains first name of user', () => {
```


here for example...